deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -41,30 +46,44 @@ public abstract class OpenApiGeneratorCodeGenBase implements CodeGenProvider {
 
     private static final String DEFAULT_PACKAGE = "org.openapi.quarkus";
 
+    /**
+     * The input base directory from <pre>src/main<pre> directory. Ignored if INPUT_BASE_DIR is specified.
+     **/
     @Override
     public String inputDirectory() {
         return "openapi";
     }
 
+    @Override
+    public boolean shouldRun(Path sourceDir, Config config) {
+        String inputBaseDir = getInputBaseDirRelativeToModule(sourceDir, config);
+        if (inputBaseDir != null && !Files.isDirectory(Path.of(inputBaseDir))) {
+            throw new OpenApiGeneratorException(String.format("Invalid path on %s: %s", INPUT_BASE_DIR, inputBaseDir));
+        }
+        return inputBaseDir != null || Files.isDirectory(sourceDir);
+    }
+
     @Override
     public boolean trigger(CodeGenContext context) throws CodeGenException {
         final Path outDir = context.outDir();
-        final Path openApiDir = context.inputDir();
-        final List<String> ignoredFiles = context.config()
-                .getOptionalValues("quarkus.openapi-generator.codegen.ignore", String.class).orElse(List.of());
+        String inputBaseDir = getInputBaseDirRelativeToModule(context.inputDir(), context.config());
+        final Path openApiDir = inputBaseDir != null ? Path.of(inputBaseDir) : context.inputDir();
```

Comment:
I think is better to cahnge the value returned by context.inputDir by overriding inputDirectory() method and reading the property there

Reply:
not really, because this path is relative to src/main, and I want to do it relative to the module